### PR TITLE
Reducing memory footprint by excluding unused rule attributes

### PR DESCRIPTION
**Pull request by brian-mcnamara**
Seeing a 2/3 to 3/4 reduction in total memory used.

### CODE REVIEW

**Review by plaird — APPROVED** [review]
Nice enhancement.

File: tests/com.salesforce.bazel.eclipse.core.tests/src/com/salesforce/bazel/eclipse/core/tests/it/Workspace001Test_BasicModelAssumptions.java [hunk 1]
```diff
@@ -84,7 +85,12 @@ void test0010_module1() throws Exception {
         var ruleAttributes = module1.getRuleAttributes();
         assertNotNull(ruleAttributes);
 
-        assertThat(ruleAttributes.getStringList("deps"), hasItems("//module2:module2", "//module3:module3",
-            "@com_google_guava//jar:jar", "//module1:mybuilder_sources"));
+        assertThat(
+            ruleAttributes.getStringList(BazelRuleAttribute.SRCS),
```
Comment (AdwaitChavan) [inline tests/com.salesforce.bazel.eclipse.core.tests/src/com/salesforce/bazel/eclipse/core/tests/it/Workspace001Test_BasicModelAssumptions.java:89]: As this previously used `"deps"`, should `BazelRuleAttribute.DEPS` be used now?

File: bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/JavaAspectsClasspathInfo.java [hunk 2]
```diff
@@ -66,6 +66,8 @@
 import com.salesforce.bazel.eclipse.core.model.discovery.classpath.AccessRule;
 import com.salesforce.bazel.eclipse.core.model.discovery.classpath.ClasspathEntry;
 import com.salesforce.bazel.sdk.command.BazelBuildWithIntelliJAspectsCommand;
+import com.salesforce.bazel.sdk.command.querylight.BazelRuleAttribute;
+import com.salesforce.bazel.sdk.model.BazelLabel;
```
Comment (AdwaitChavan) [inline bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/JavaAspectsClasspathInfo.java:70]: This import might not be needed here, based on the changes in this file.

File: bundles/com.salesforce.bazel.sdk/src/com/salesforce/bazel/sdk/command/querylight/BazelRuleAttribute.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ *      Salesforce - adapted from M2E, JDT or other Eclipse project
+ */
+package com.salesforce.bazel.sdk.command.querylight;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Rule attribute names we need
```
Comment (AdwaitChavan) [inline bundles/com.salesforce.bazel.sdk/src/com/salesforce/bazel/sdk/command/querylight/BazelRuleAttribute.java:21]:
```suggestion
 * Stores minimal Bazel Rule Attribute names needed to work with Bazel based Java Projects. Helps reduce memory usage by restricting to only the necessary Bazel Rule Attribute names.
```

**Review by AdwaitChavan — APPROVED** [review]

**Review by guw — APPROVED** [review]

**Pull request merged**